1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,6 +1,49 @@
-function setAlarm() {}
+// these 4 are globally scoped because they will be re-used across various functions
 
-// DO NOT EDIT BELOW HERE
+const timeInput = document.querySelector("#alarmSet");
+const timeRemaining = document.querySelector("#timeRemaining");
+
+let time;
+let intervalID;
+
+function setAlarm() {
+  console.log("setAlarm function ran");
+
+  time = timeInput.value;
+  timeRemaining.textContent = `Time Remaining: ${formatTime(time)}`;
+
+  intervalID = setInterval(() => {
+    console.log("setInterval function ran");
+    time--;
+    console.log("Timer is:",time)
+    timeRemaining.textContent = `Time Remaining: ${formatTime(time)}`;
+    if (time === 0) {
+      clearInterval(intervalID);
+      playAlarm();
+    }
+  }, 1000);
+}
+
+function cancelAlarm() {
+  console.log("cancelAlarm function ran");
+
+  clearInterval(intervalID);
+  stopAlarm();
+  time = 0;
+  timeRemaining.textContent = `Time Remaining: ${formatTime(time)}`;
+}
+
+function formatTime(time) {
+  const minutes = Math.floor(time / 60);
+  console.log("formatTime minutes:", minutes);
+  const seconds = time % 60;
+  console.log("formatTime seconds:", seconds);
+  const formattedMinutes = minutes < 10 ? `0${minutes}` : `${minutes}`;
```

Comment:
Nice use of single line condition. FYI there is a built-in function of string type in Javascript `padStart()` that helps give the prefix when needed, you can check here https://www.w3schools.com/jsref/jsref_string_padstart.asp.
You may want to know how to convert a number into a string, you can check here https://w3schools.com/jsref/jsref_tostring_number.asp